registry/storage/registry.go

```diff
@@ -13,24 +14,27 @@ type registry struct {
 	driver    storagedriver.StorageDriver
 	pm        *pathMapper
 	blobStore *blobStore
+	redis     *redis.Pool
 }
 
 // NewRegistryWithDriver creates a new registry instance from the provided
 // driver. The resulting registry may be shared by multiple goroutines but is
 // cheap to allocate.
-func NewRegistryWithDriver(driver storagedriver.StorageDriver) distribution.Registry {
-	bs := &blobStore{}
+func NewRegistryWithDriver(driver storagedriver.StorageDriver, redis *redis.Pool) distribution.Registry {
```

Comment:
What would that interface look like? Make sure to think about the implementation, since I've already tried the simple approaches.